### ISSUE TITLE
Defaulting to the best block is insecure

### ISSUE THREAD

**Issue by tomaka** [paritytech/capi#239]
I see several examples showing that it is possible to call `read()` in order to read a storage value from, as far as I understand, the best block.

If the JSON-RPC server is a light node, then this is completely insecure. **A light client doesn't offer any guarantee about the validity of the best block**. Only the finalized block can be trusted. This is a fundamental property of Substrate-based light clients and will not change in the future.

For this reason, I don't think that it is a good idea to default to reading from the best block.

However, users might still be interested in reading a storage value or watching a storage value of the best block in order to show the changes that will likely be part of the finalized block.
In other words, a storage value in the best block should be seen as "pending changes that might end up not being applied" while storage values in the finalized block should be seen as "has actually made it into the chain".

**Comment by harrysolovay** [comment]
Thank you identifying this most necessary change! Will address in the next few weeks.
